### ISSUE TITLE
[a11y] Tooltips not displayed correctly with low vision settings

### ISSUE THREAD

**Issue by dpersing** [Shopify/polaris-react#793]
## Issue summary

The [Tooltip](https://polaris.shopify.com/components/overlays/tooltip#navigation) has some issues that may cause problems for low vision users:

- When the tooltip is viewed in a Windows High Contrast theme, the shape of the tooltip is not visible, since its background styles are removed. Depending on the context, it may be difficult for users to see the relationship between the trigger and the tooltip.
- When the Tooltip is viewed in Firefox with text resized by 200%, the tooltip is not visible when the button is hovered/focused. This seems to be related to the fixed height of the example's iframe container. Increasing the size of the container makes the tooltip visible. (Visually, resizing text only generally triggers smaller breakpoints but can bump up against fixed containers sizes and cause content to get cut off.)

![Tooltip standard appearance](https://user-images.githubusercontent.com/1462085/41864476-02e4e8c8-785f-11e8-9b7b-bce2d6a3cf10.png)

![Tooltip in Windows High Contrast with a dark theme](https://user-images.githubusercontent.com/1462085/41864452-ebfd73b4-785e-11e8-913e-55f8dedb8d2e.png)

![Tooltip in Windows High Contrast with a white theme](https://user-images.githubusercontent.com/1462085/41864465-f8800444-785e-11e8-9092-da0d51d8444c.png)

#### Tooltip appearance (CSS)

```
.Polaris-Tooltip__Tip {
    position: absolute;
    top: 0;
    left: 50%;
    transform: translateX(-50%) rotate(45deg);
    height: 1.4rem;
    width: 1.4rem;
    background: #212b36;
    box-shadow: 0 0 0 1px rgba(6,44,82,.1), 0 2px 16px rgba(33,43,54,.08);
}

.Polaris-Tooltip__Wrapper {
    display: -ms-flexbox;
    display: flex;
    background-color: #212b36;
    color: #fff;
}
```

#### Iframe (HTML)

```
<iframe title="Rendered example" src="/demo" height="210" frameborder="0"></iframe>
```

#### High Contrast

1. Access [Tooltip](https://polaris.shopify.com/components/overlays/tooltip#navigation) using Edge on Windows 10.
1. Use the mouse or the keybaord to activate the tooltip and do a visual inspection.
1. Turn on any [High Contrast theme in Windows settings](https://support.microsoft.com/en-us/help/13862/windows-use-high-contrast-mode) and redo the visual inspection.

#### Text resizing

1. Open Firefox on macOS or Windows.
1. Set [Firefox to resize text only](https://support.mozilla.org/en-US/kb/font-size-and-zoom-increase-size-of-web-pages#w_how-to-only-change-the-size-of-the-text).
1. Navigate to  [Tooltip](https://polaris.shopify.com/components/overlays/tooltip#navigation).
1. Use the mouse or the keyboard to activate the tooltip.

### Expected

- Understanding the structure and information of the tooltip will not rely on color alone.
- The tooltip will be viewable when text alone is resized.

### Actual

- The tooltip shape is not visible in Windows High Contrast, since the background styles are removed.
- The tooltip is not visible when text is resized by 200% because its parent container has a fixed [px] height.

## Recommendations

- Use `-ms-high-contrast` to create a tooltip style that does not rely on color alone. Using a solid border on the tooltip container is recommended. (It's okay if it doesn't look beautiful; it just needs to be clear and readable.)
- Make the example `<iframe>` slightly larger to accommodate how containers shift with text resizing, or refrain from setting a fixed height on containers that may impact `overflow` text. This is more of an issue with the specific presentation than with the tooltip, but it's good to take into account how the tooltip might behave inside fixed-sized containers in general.

## Specifications

- **Component:** Tooltip
- **WCAG SCs:** [1.4.1 Use of Color (Level A)](https://www.w3.org/TR/WCAG21/#use-of-color), 
- **Testing done with:**
  - Text-only resizing (Firefox)
  - Windows High Contrast/Edge/Windows 10

**Comment by ruairiphackett** [comment]
@dpersing 

### Windows high contrast

How do you feel abut a 2px (`border-width(thick)`) solid border with the same colour as the text:

![https://screenshot.click/2019-05-02_12-22-55.png](https://screenshot.click/2019-05-02_12-22-55.png)

### Firefox

I wasn't able to reproduce the Firefox issue, I wonder if it has been addressed by changes to Firefox since this issue was created? (Or else I'm doing something wrong)

Here's what I see with text zoomed to 200%
![https://screenshot.click/2019-05-02_12-48-56.gif](https://screenshot.click/2019-05-02_12-48-56.gif)

It does act strange when I resize the window however, not sure if that's related to what you mentioned:

![https://screenshot.click/2019-05-02_12-51-58.gif](https://screenshot.click/2019-05-02_12-51-58.gif)

**Comment by dpersing** [comment]
Hi @ruairiphackett! The High Contrast border looks great to me.

For the Firefox issue, it looks like it's changed a bit, but I'm still getting some layout problems when text-only resizing happens. (This isn't the default resizing setting, but is technically a WCAG requirement. I should have linked to the detail for [changing the resize setting](https://support.mozilla.org/en-US/kb/font-size-and-zoom-increase-size-of-web-pages#w_how-to-only-change-the-size-of-the-text).)

The tooltip text is visible on hover/focus, but it's overflowing the tooltip container. This appears to be the same on Windows and macOS.

<img width="1094" alt="Tooltip overflow!" src="https://user-images.githubusercontent.com/1462085/57091371-e221de80-6cbd-11e9-83e8-ba6273c69957.png">
